### ISSUE TITLE
Can not set reminder for private channels

### ISSUE THREAD

**Issue by JanMalte** [scottleedavis/mattermost-plugin-remind#188]
#### Summary
Reminders for private channels is not supported. I can set the reminder, but it never gets posted.

#### Steps to reproduce
1. Create a private channel
2. Set a reminder for the channel
3. Nothing happens at the set time

#### Expected behavior
The reminder should be posted to the private channel

#### Observed behavior (that appears unintentional)
The reminder is set, but doesn't get posted
![Screenshot_20200617_162847](https://user-images.githubusercontent.com/544986/84910812-a4496a80-b0b7-11ea-9db0-c76b81e7af2c.png)

#### Server configuration
**Mattermost version:**
Mattermost Version: 5.21.0
Database Schema Version: 5.21.0
Database: postgres

**Remind version:**
Remind Bot (com.github.scottleedavis.mattermost-plugin-remind - 0.4.4)

<details>
<summary>Mattermost server logs</summary>
{"level":"error","ts":1592403796.264903,"caller":"mlog/sugar.go:23","msg":"failed to find target user @channel","plugin_id":"com.github.scottleedavis.mattermost-plugin-remind"}
{"level":"error","ts":1592403933.994252,"caller":"mlog/sugar.go:23","msg":"fail to get channel SqlChannelStore.GetByName: Channel does not exist, teamId=hthgqqkffbn6ffbjx3sgqma6mh, name=it-intern","plugin_id":"com.github.scottleedavis.mattermost-plugin-remind"}
</details>

**Comment by olivierlambert** [comment]
Same problem here :)

**Comment by micabot** [comment]
Same here! Any update on this?

**Comment by dongnak** [comment]
Same here.

**Comment by scastillo** [comment]
I haven't double-checked the implementation but it seems it may be related to this: https://github.com/mattermost/mattermost-server/issues/16245

I'm very interested in solving this so if anyone is working on a workaround I would really appreciate if you let me know. I can collab to get it done.

**Comment by olivierlambert** [comment]
The only workaround I found is to use it for… public channels :(

@scastillo keep me posted if you find anything!

**Comment by scottleedavis** [comment]
Happy to help resolve this.

**Comment by Hespian** [comment]
Is there any update on this? We're having the same problem and all our channels are private so we can't use channel reminders at all (except for Town Square of course)

**Comment by tls-fledoux** [comment]
Hi all. I found a solution for this issue. You just have to add the bot account ( @remindbot created during installation) on your private channel like a regular user. ( "Invite People" on your server + "Add new members" on the private channel ). Then just use the /remind ~privateChannel "message" command. Works fine for me :)

**Comment by scottleedavis** [comment]
thank you @tls-fledoux .  closing this as fixed.

**Comment by olivierlambert** [comment]
I don't have such bot in my installation 🤔 
I will reinstall the plugin.

**Comment by JanMalte** [comment]
I could not add the user `remindbot` or `@remindbot` as they are not available in the "Add user" popup.
They are not real users but `Bot Accounts`

**Comment by tls-fledoux** [comment]
Here is the procedure I followed : In the dropdown menu on the top-left corner, select "Invite People". (![1](https://user-images.githubusercontent.com/85942219/127826243-16aa32b1-45a5-4a67-ad87-2f947e44452d.png)) Then I was able to "invite" the bot account on the server (auto-accepted) (![2](https://user-images.githubusercontent.com/85942219/127826307-0bf8e448-e394-420c-a0e7-a513dcbf89df.png)
). Then the bot account appears in the "Add user" menu on channels.
